sql/core/src/main/scala/org/apache/spark/sql/execution/command/tables.scala

```diff
@@ -617,10 +617,55 @@ case class DescribeTableCommand(
   }
 
   private def append(
-      buffer: ArrayBuffer[Row], column: String, dataType: String, comment: String): Unit = {
+    buffer: ArrayBuffer[Row], column: String, dataType: String, comment: String): Unit = {
     buffer += Row(column, dataType, comment)
   }
 }
+/**
+ * Command can of the following form.
+ * {{{
+ *   DESCRIBE [EXTENDED|FORMATTED] table_name partitionSpec?;
+ * }}}
+ */
+case class DescribeTableCommand(
+    tableIdent: TableIdentifier,
+    partitionSpec: TablePartitionSpec,
+    isExtended: Boolean)
+  extends DescribeTableCommandBase {
+
+  override def run(sparkSession: SparkSession): Seq[Row] = {
+    val result = new ArrayBuffer[Row]
+    describeTable(sparkSession, tableIdent, partitionSpec, isExtended, result)
+    result
+  }
+}
+
+/**
+ * Command can of the following form.
+ * {{{
+ *   DESCRIBE [QUERY] query_statement
+ * }}}
+ */
+case class DescribeQueryCommand(query: LogicalPlan)
+  extends DescribeTableCommandBase {
+
+  override def run(sparkSession: SparkSession): Seq[Row] = {
+    val result = new ArrayBuffer[Row]
+    query match {
+      case p if p.collectFirst { case _: InsertIntoTable | _: InsertIntoDir => true }.isDefined =>
+        throw new AnalysisException("Describe command is not supported for insert statements.")
+      case u: UnresolvedRelation =>
```

Comment:
how can this happen?

Reply:
@cloud-fan Can you please check out my response at [link](https://github.com/apache/spark/pull/23883#discussion_r260099018)

Reply:
So this should belong to describe table, but our parser is not able to distinguish it.

shall we switch the order of `DESCRIBE TABLE` and `DESCRIBE QUERY` in the parser?

Reply:
@cloud-fan Good idea Wenchen. Let me try.

Reply:
@cloud-fan Thanks a lot. I have incorporated your suggestion. Please take a look when you get a chance. Just waiting for your decision on whether to raise and error or return empty schema.